src/JwtCredentialManager.js

```diff
@@ -0,0 +1,84 @@
+/* @flow */
+import type { ICredentialsManager } from './CredentialManager';
+
+type Seconds = number;
+
+export type JwtCredentials = {
+  exp: Seconds,
+};
+
+export type JwtCredentialsManagerConfig<TCredentials> = {|
+  enableTokenRenewal: ?boolean, // default: true
+  gracePeriod: Seconds,
+  initialCredentials: TCredentials,
+|};
+
+const SECONDS_TO_MS = 1000;
+
+class JwtCredentialsManager<TCreds: JwtCredentials>
+  implements ICredentialsManager<TCreds> {
+  config: JwtCredentialsManagerConfig<TCreds>;
+  credentials: TCreds;
+  token: ?string;
+  renewTimer: ?TimeoutID;
+
+  constructor(config: JwtCredentialsManagerConfig<TCreds>) {
+    this.config = config;
+    this.token = null;
+    this.credentials = config.initialCredentials;
+  }
+
+  getCredentials(): TCreds {
+    return this.credentials;
+  }
+
+  getCredentialsFromAuthorization(
+    // eslint-disable-next-line no-unused-vars
+    authorization: string,
+  ): TCreds | Promise<TCreds> {
+    throw new Error('Not implemented');
+  }
```

Comment:
not sure the abstract method is better than a ctor function like it used to be...

Reply:
if flow had abstract class support, this would be perfect